scipy/spatial/kdtree.py

```diff
@@ -236,174 +222,88 @@ class KDTree(object):
     sort of calculation.
 
     """
-    def __init__(self, data, leafsize=10):
-        self.data = np.asarray(data)
-        if self.data.dtype.kind == 'c':
-            raise TypeError("KDTree does not work with complex data")
 
-        self.n, self.m = np.shape(self.data)
-        self.leafsize = int(leafsize)
-        if self.leafsize < 1:
-            raise ValueError("leafsize must be at least 1")
-        self.maxes = np.amax(self.data,axis=0)
-        self.mins = np.amin(self.data,axis=0)
+    class node:
+        @staticmethod
+        def __new__(cls, ckdtree_node=None):
```

Comment:
I suggest avoid overriding the `__new__` magic method, which can lead to tricky situations.

It can be more readable to use a static Create method that directly creates instances of the desired type with similar amount of code.

Reply:
Renamed to `_create`.